src/irr/asset/CGLTFLoader.cpp

```diff
@@ -50,6 +50,203 @@ namespace irr
 				cpuBuffers.emplace_back() = core::smart_refctd_ptr<ICPUBuffer>(cpuBuffer);
 			}
 
+			core::vector<core::smart_refctd_ptr<CCPUMesh>> cpuMeshes;
+			for (auto& glTFnode : glTF.nodes) 
+			{
+				auto cpuMesh = cpuMeshes.emplace_back();
+
+				for (auto& glTFprimitive : glTFnode.glTFMesh.primitives) 
+				{
+					typedef std::remove_reference<decltype(glTFprimitive)>::type SGLTFPrimitive;
+					
+					auto meshBuffer = core::make_smart_refctd_ptr<asset::ICPUMeshBuffer>();
+
+					auto getMode = [&](uint32_t modeValue) -> E_PRIMITIVE_TOPOLOGY
+					{
+						switch (modeValue)
+						{
+							case SGLTFPrimitive::SGLTFPT_POINTS:
+								return EPT_POINT_LIST;
+							case SGLTFPrimitive::SGLTFPT_LINES:
+								return EPT_LINE_LIST;
+							case SGLTFPrimitive::SGLTFPT_LINE_LOOP:
+								return EPT_LINE_LIST_WITH_ADJACENCY; // check it
+							case SGLTFPrimitive::SGLTFPT_LINE_STRIP:
+								return EPT_LINE_STRIP;
+							case SGLTFPrimitive::SGLTFPT_TRIANGLES:
+								return EPT_TRIANGLE_LIST;
+							case SGLTFPrimitive::SGLTFPT_TRIANGLE_STRIP:
+								return EPT_TRIANGLE_STRIP;
+							case SGLTFPrimitive::SGLTFPT_TRIANGLE_FAN:
+								return EPT_TRIANGLE_STRIP_WITH_ADJACENCY; // check it
+						}
+					};
+
+					const E_PRIMITIVE_TOPOLOGY primitiveTopology = getMode(glTFprimitive.mode.value());
+
+					if (glTFprimitive.indices.has_value())
+					{	
+						auto& glTFIndexAccessor = glTFprimitive.accessors["INDEX"];
+
+						// TODO
```

Comment:
just remember index buffers cannot have strides with gaps, if you encounter something like this, issue a warning and deinterleave the index buffer into a copy.

Reply:
so we want to make sure index buffer is put in separate buffer?

Reply:
no, not unless the stride of the index buffer's accessor does not equal the size of the index type (i.e. there would be gaps)